### DISCUSSION TITLE
Open source alternative for the vscode cpptools

### DISCUSSION THREAD

**Discussion by niekdomi** [VSCodium/vscodium#1971] [Q&A]
Hi,

Since VSCodium uses open-vsx as its marketplace, only open-source plugins are supported (as far as I know). Therefore, my question is: is there an open-source alternative to the [vscode cpptools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) for debugging and compiling C/C++ code?

**Comment by ghost** [comment]
Try the LLVM family, clangd + CodeLLDB + Clang-format (not an vsx extension though, it's included in the LLVM suite, I think).

And, you can also change the `product.json` as shown in previous issues.

Have a nice day. :)
